electron/js/lib/openGraph.js

```diff
@@ -41,21 +42,37 @@ const fetchImageAsBase64 = url => {
 };
 
 const fetchOpenGraphData = url => {
-  const isHtmlContentTypePromise = new Promise(resolve => {
-    const headerRequest = request.head(url);
-    headerRequest.on('response', response => {
-      const isHtmlContentType =
-        response.headers['content-type'] && response.headers['content-type'].match(/.*text\/html/);
-      resolve(isHtmlContentType);
+  const CONTENT_TYPE_LIMIT = 1e7; // 10MB
+  const parsedUrl = urlUtil.parse(url);
+  const normalizedUrl = parsedUrl.protocol ? parsedUrl : urlUtil.parse(`http://${url}`);
+
+  return new Promise((resolve, reject) => {
+    const getContentRequest = request.get(urlUtil.format(normalizedUrl), (error, response, body) => {
+      return error ? reject(error) : resolve(body);
     });
-  });
 
-  return isHtmlContentTypePromise.then(isHtmlContentType => {
-    if (isHtmlContentType) {
-      return new Promise((resolve, reject) => openGraph(url, (error, meta) => (error ? reject(error) : resolve(meta))));
-    }
-    throw new Error('Unhandled format for open graph generation');
-  });
+    getContentRequest.on('response', ({headers}) => {
+      const contentType = headers['content-type'] || '';
+      const contentLength = parseInt(headers['content-length'] || '0', 10);
+
+      const isHtmlContentType = contentType.match(/.*text\/html/);
+      const isTooLarge = contentLength > CONTENT_TYPE_LIMIT;
```

Comment:
If you really want to check the content type and its length then you need to restore the HEAD request otherwise those checks only happen at the end.

Reply:
The `on('response')` event is actually called before any content is downloaded (basically when the headers are here)

Reply:
Headers are part of the downloaded content. I think you need a HEAD request.

Reply:
Another approach if we don't want to do 2 requests would be extension blacklisting but it's far from perfect.

Reply:
I checked, this function is called before any `data` event occurs. We are pretty safe on that side :+1:

Reply:
You can find the lyfecycle of a request here https://nodejs.org/api/http.html#http_http_request_options_callback

Actually if the body is empty, no `data` event will be emitted.